astropy/coordinates/sky_coordinate.py

```diff
@@ -1003,6 +1007,105 @@ def to_string(self, style='decimal', **kwargs):
+        with it. In case of a column name or metadata conflict data from the
+        |SkyCoord| is preferred.
+
+        Parameters
+        ----------
+        table : |Table|, optional
+            A table to which the data from the |SkyCoord| is added.
+        column_conflicts : str ('warn'|'error'|'silent')
+            How to handle column name conflicts. Has no effect if ``table`` is
+            not provided.
+        metadata_conflicts : str ('warn'|'error'|'silent')
+            How to handle metadata conflicts. Has no effect if ``table`` is not
+            provided.
+
+        Returns
+        -------
+        |QTable|
+            A |QTable| containing the data of this |SkyCoord|, merged with
+            ``table`` if it is provided.
+
+        Examples
+        --------
+        >>> sc = SkyCoord(ra=[40, 70]*u.deg, dec=[0, -20]*u.deg,
+        ...               obstime=Time([2000, 2010], format='jyear'))
+        >>> t =  sc.to_table()
+        >>> t
+        <QTable length=2>
+           ra     dec   obstime
+          deg     deg
+        float64 float64   Time
+        ------- ------- -------
+           40.0     0.0  2000.0
+           70.0   -20.0  2010.0
+        >>> t.meta
+        {'representation_type': 'spherical', 'frame': 'icrs'}
+        >>> t = Table({'a': [3, 7]}, meta={'b': 'c'})
+        >>> t = sc.to_table(t)
+        >>> t
+        <QTable length=2>
+          a      ra     dec   obstime
+                deg     deg
+        int64 float64 float64   Time
+        ----- ------- ------- -------
+            3    40.0     0.0  2000.0
+            7    70.0   -20.0  2010.0
+        >>> t.meta
+        {'b': 'c', 'representation_type': 'spherical', 'frame': 'icrs'}
+        """
+        self_as_dict = self.info._represent_as_dict()
+        tabledata = {}
+        metadata = {}
+        # Record attributes that have the same length as self as columns in the
+        # table, and the other attributes as table metadata.  This matches
+        # table.serialize._represent_mixin_as_column().
+        for key, value in self_as_dict.items():
+            if getattr(value, 'shape', ())[:1] == (len(self),):
+                tabledata[key] = value
+            else:
+                metadata[key] = value
+        sc_as_table = QTable(tabledata, meta=metadata)
+        if table is None:
+            return sc_as_table
+        elif not isinstance(table, Table):
+            raise TypeError("The argument `table` must be a Table")
+        if len(sc_as_table) != len(table):
+            raise ValueError("The input table has a different length than the "
+                             "SkyCoord")
+        if column_conflicts not in ('silent', 'warn', 'error'):
+            raise ValueError("The `column_conflicts` argument must be "
+                             "'silent', 'warn' or 'error' (got "
+                             f"'{column_conflicts}' instead)")
+        if column_conflicts != 'silent':
```

Comment:
Is this done manually in the table code? E.g. the metadata merge handler uses a registry of functions and abstracts a bit over these details. If column_conflicts in the ``astropy.table`` code is changed, that won't automatically be supported here.

Reply:
The table operations solve column name conflicts by renaming the conflicting columns, and the user can provide a pattern for the renamed columns (see here: https://docs.astropy.org/en/stable/table/operations.html#column-renaming). This function replaces the old column from the table with a new one from the `SkyCoord` instead. This means that `SkyCoord.to_table()` could be easily used for something like updating coordinates in a table after epoch propagation, which is certainly useful. Replacing the very simple code here with something that tries to make use of the machinery in table operations doesn't seem like a good idea.

Reply:
All this is great. My original concern is simply that the functionality of column conflicts is mirrored, but not any code itself. Improvements/extensions elsewhere would need to be manually duplicated here. Now this isn't a huge deal on a per-case basis — the per-case marginal cost is relatively low. I'm not an `astropy.table` maintainer and I'm not sure if there even is existing machinery, but my inclination would be to, like metadata in `astropy.utils.metadata`, have a system to handle column conflicts. One code used in multiple places. If this machinery doesn't exist, my question is moot (but might spawn a new issue as that machinery would be very out of scope).

Reply:
The way column name conflicts are handled here is completely different from how the table operations do it. The table joining functions rename the conflicting columns so that the resulting table contains both. The user can specify a pattern for this renaming, but there is no equivalent of the `column_conflicts` argument at all.

The way column name conflicts are handled here has much more in common with how metadata conflicts are handled, but trying to make use of that machinery for columns sounds quite convoluted.

If the user wishes to make use of the machinery in the table joining functions then it is possible to first convert the `SkyCoord` to a table and then combine the tables using the existing functions.

Reply:
> there is no equivalent of the column_conflicts argument at all.

Ok. Then this code is AOK. when perusing the code I saw the similarity and wondered.

*Resolved*